### ISSUE TITLE
☂️ Avatar component

### ISSUE THREAD

**Issue by patricklafrance** [gsoft-inc/sg-orbit#535]
## Description

An avatar component doesn't have to match any specific accessibility specifications other than the classic `alt` attribute when it's an `img` tag. However, since ShareGate is mostly about b2b applications in the Microsoft ecosystem, an avatar is such a common component that it does make sense to have it in Orbit.

Furthermore, our Menu component support an embedded Avatar in an item.

A few examples of Orbit avatar:

![image](https://user-images.githubusercontent.com/794579/115620501-3ebf9f00-a2c3-11eb-86ca-63fabda159ec.png)

![image](https://user-images.githubusercontent.com/794579/115620543-497a3400-a2c3-11eb-9b71-28e95ae76e58.png)

![image](https://user-images.githubusercontent.com/794579/115620572-5139d880-a2c3-11eb-8362-7ae6651d29ae.png)

![image](https://user-images.githubusercontent.com/794579/115620590-572fb980-a2c3-11eb-9809-2455e8fed2b8.png)

![image](https://user-images.githubusercontent.com/794579/115620644-6e6ea700-a2c3-11eb-9979-ab0505357716.png)

![image](https://user-images.githubusercontent.com/794579/115620707-87775800-a2c3-11eb-9166-abdc5ee7c2af.png)

Inspiration:
- https://paste.twilio.design/components/avatar
- https://chakra-ui.com/docs/media-and-icons/avatar

Figma:
- https://www.figma.com/file/VATs1Zu2vEXqIBicdwP3fI/Orbit---UIKit?node-id=7147%3A53623

## Specifications

- Support local image file & remote image
- Support textual avatar (initials)
- Textual avatar colors should match O365
- When a remote image avatar failed to load, fallback to textual avatar
- Support avatar group (with stacking)
- Support size "xs", "sm", "md", "lg", "xl" "2xl"

## To do:

- [x] Design
- [ ] Development
- [ ] Documentation
